scripts/build.sh

```diff
@@ -0,0 +1,147 @@
+
+function main() {
+  local name token
+  token=""
+
+  while [[ "${#}" != 0 ]]; do
+    case "${1}" in
+      --help|-h)
+        shift 1
+        usage
+        exit 0
+        ;;
+
+      --name|-n)
+        name="${2}"
+        shift 2
+        ;;
+
+      --token|-t)
+        token="${2}"
+        shift 2
+        ;;
+
+      "")
+        # skip if the argument is empty
+        shift 1
+        ;;
+
+      *)
+        util::print::error "unknown argument \"${1}\""
+    esac
+  done
+
+  if [[ -z "${name:-}" ]]; then
+    name="testbuilder"
+  fi
+
+  # tools::install "${token}"
+  if [[ ! -f "${BUILDERDIR}/.bin/pack" ]]; then
+    experimental::pack::install "${token}"
+  fi
+
+  # Do not rebuild the builder if it already exists on the local registry
+  if [[  -z $(docker ps -aqf "name=${LOCAL_REGISTRY_NAME}") ]]; then
+    local_registry::run >/dev/null
+  else
+    echo "Local registry named ${LOCAL_REGISTRY_NAME} is already running"
+    registry_images=$(curl -X GET "localhost:5000/v2/_catalog" | jq -r '.repositories')
+    if echo "${registry_images}" | grep -q "${name}"; then
+      echo "A builder image named localhost:5000/${name} on the local registry already exists"
+      exit 0
+    fi
+  fi
+
+  name="localhost:5000/${name}"
+  builder::create "${name}"
+}
+
+function usage() {
+  cat <<-USAGE
+build.sh [OPTIONS]
+
+Builds the multi-arch builder using a local docker registry
+
+OPTIONS
+  --help        -h         prints the command usage
+  --name <name> -n <name>  sets the name of the builder that is built
+  --token <token>          Token used to download assets from GitHub (e.g. jam, pack, etc) (optional)
+USAGE
+}
+
+function tools::install() {
+  local token
+  token="${1}"
+
+  util::tools::pack::install \
+    --directory "${BUILDERDIR}/.bin" \
+    --token "${token}"
+}
+
+# TODO: when an official pack release comes out that supports multi-arch, use that
+#  Until then, use an experimental pack version we've saved to a shared location
+function experimental::pack::install() {
+  echo "Installing pack experimental"
+
+  local artifact_uri
+  os=$(util::tools::os)
+  arch=$(util::tools::arch)
+  mkdir -p "${BUILDERDIR}/.bin"
+
+  if [[ ${arch}  == "amd64" ]]; then
+    if [[ ${os}  == "linux" ]]; then
+      # linux + amd64
+      artifact_uri="https://api.github.com/repos/buildpacks/pack/actions/artifacts/1306592515/zip"
+    elif [[ ${os}  == "darwin" ]]; then
+      # darwin + amd64
+      artifact_uri="https://api.github.com/repos/buildpacks/pack/actions/artifacts/1306588326/zip"
+    fi
+
+    curl -L -H "Accept: application/vnd.github+json" -H "Authorization: Bearer ${token}" "${artifact_uri}" \
+      -o "${BUILDERDIR}/.bin/pack.zip"
+    unzip "${BUILDERDIR}/.bin/pack.zip" -d "${BUILDERDIR}/.bin"
+
+  elif [[ ${arch}  == "arm64" ]]; then
+    pushd /tmp
+      git clone https://${token}@github.com/buildpacks/pack.git
+      cd pack
+      git fetch origin pull/2086/head:pr-2086
```

Comment:
be careful with that: the day the PR is closed / merged and the branch is deleted it could away

That said, it could also serve as a reminder that day that you need to use the "officially" released `pack`

Reply:
Good point. Do you think it would be better to build it and store it somewhere instead?

Reply:
you could put it on box.com just like the x86, sure - but we'll need to replace it eventually with the real one once it's released - either you checkout from git or download from box.com
As you wish!

Reply:
I think I'll leave this for now with the intent to consume the official pack release as soon as it's available.